### PR TITLE
Refactored base code

### PR DESCRIPTION
**Pull request by ProstoMaxim**
Refactored base code

### CODE REVIEW

**Comment by ProstoMaxim** [comment]
Any ideas why this is failing?

**Comment by ashwoods** [comment]
The flake8 tests are failing, stricter settings were set recently. 
I'm a bit hesitant to merge style-only PRs, but we are planning a cleanup sprint and your PR is pretty clean. thx!

File: raven/utils/stacks.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def _getitem_from_frame(f_locals, key, default=None):
     """
     try:
         return f_locals[key]
-    except Exception:
```
Comment (ashwoods) [inline raven/utils/stacks.py]: a bare exception is <> an except Exception, as it doesn't catch i.e. system existing exceptions.

Comment (ProstoMaxim) [reply raven/utils/stacks.py]: Fixed

File: raven/utils/stacks.py [hunk 2]
```diff
@@ -135,23 +135,22 @@ def iter_stack_frames(frames=None):
...
-        yield frame, lineno
+        if not _getitem_from_frame(f_locals, '__traceback_hide__'):
+            yield frame, lineno
 
 
 def get_frame_locals(frame, transformer=transform, max_var_size=4096):
     f_locals = getattr(frame, 'f_locals', None)
     if not f_locals:
-        return None
+        return
```
[the remark this answers is not in the document]
Comment (ProstoMaxim) [reply raven/utils/stacks.py]: Fixed

File: raven/utils/stacks.py [hunk 3]
```diff
@@ -135,23 +135,22 @@ def iter_stack_frames(frames=None):
...
 
     if not isinstance(f_locals, dict):
         # XXX: Genshi (and maybe others) have broken implementations of
         # f_locals that are not actually dictionaries
         try:
             f_locals = to_dict(f_locals)
-        except Exception:
```
[the remark this answers is not in the document]
Comment (ProstoMaxim) [reply raven/utils/stacks.py]: Fixed

**Comment by ProstoMaxim** [comment]
Done

**Pull request merged**